src/providers/ad/ad_pac_common.c

```diff
@@ -24,9 +24,186 @@
 #include "providers/ad/ad_pac.h"
 #include "util/util.h"
 
+static bool
+compare_sid_with_dom_sid_and_rid(const struct dom_sid *sid,
+                                 const struct dom_sid *dom,
+                                 uint32_t rid)
+{
+    size_t c;
+
+    if (sid == NULL || dom == NULL || rid == 0) {
+        return false;
+    }
+
+    if (sid->sid_rev_num != dom->sid_rev_num) {
+        return false;
+    }
+
+    for (c = 0; c < 6; c++) {
+        if (sid->id_auth[c] != dom->id_auth[c]) {
+            return false;
+        }
+    }
+
+    if (sid->num_auths != dom->num_auths + 1) {
+        return false;
+    }
+
+    for (c = 0; c < sid->num_auths; c++) {
+        if (c == dom->num_auths) {
+            if (sid->sub_auths[c] != rid) {
+                return false;
+            }
+        } else {
+            if (sid->sub_auths[c] != dom->sub_auths[c]) {
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+static errno_t
+check_logon_info_upn_dns_info(struct PAC_LOGON_INFO *logon_info,
+                              struct PAC_UPN_DNS_INFO *upn_dns_info)
+{
+    char *delim;
```

Comment:
`delim` and both function arguments can be `const`.

Reply:
fixed